tensorflow/core/kernels/dynamic_partition_op_gpu.cu.cc

```diff
@@ -427,10 +438,14 @@ class DynamicPartitionOpGPU : public AsyncOpKernel {
     // each index appears in partitions. The distinct indices are stored
     // in unique_out, while the count is stored in aggregates_out.
     // The total number of distinct indices is stored in num_runs.
-    gpuprim::DeviceReduce::ReduceByKey(
-        cub_temp_storage.flat<int8>().data(), temp_storage_bytes, keys_in_ptr,
-        unique_out_it, values_in, aggregates_out_it, num_runs_ptr, reduction_op,
-        N, cu_stream);
+    gpuResult = gpuprim::DeviceReduce::ReduceByKey(
+                cub_temp_storage.flat<int8>().data(), temp_storage_bytes, keys_in_ptr,
+                unique_out_it, values_in, aggregates_out_it, num_runs_ptr, reduction_op,
+                N, cu_stream);
+
+    OP_REQUIRES(c, gpuResult == gpuSuccess, 
+        errors::Internal("Failed to launch copy from device to host."));
```

Comment:
errors::Internal("Failed to launch gpuprim::DeviceRadixSort::SortPairs to calculate temp_storage_bytes, status: ", cudaGetErrorString(gpuResult));

Reply:
This and the below one are for gpuprim::DeviceReduce::ReduceByKey, wondering "Failed to launch gpuprim::DeviceRadixSort::SortPairs" is also related to?

Reply:
Ah, sorry, I missed this. In that case I guess just adjust the message which method is called. It still seems to be two calls, one to calculate temporary storage requirements, and another one where temporary storage has already been allocated.